Makefile

```diff
@@ -196,17 +197,34 @@ build-image:
 run: build-image
 	@CID=$(shell docker run --net=host -d --name $(SERVER) $(IMAGE)) && echo $${CID}
 
+test-cli:
+	@for pkg in $(CLI_TEST_PACKAGES) ; do \
+		go test $$pkg ; \
+	done
+
 test-unit:
 	@for pkg in $(UNIT_TEST_PACKAGES) ; do \
 		go test $$pkg ; \
 	done
 
 test-integration:
+	@docker service rm amp-integration-test || true
+	@docker build -f Dockerfile.test -t appcelerator/amp-integration-test .
+	@docker service create --network amp-infra --name amp-integration-test --restart-condition none appcelerator/amp-integration-test
+	@n=""; \
+	while [[ $${n} == "" ]] ; do \
+		n=`docker ps -qf 'name=amp-integration'`; \
```

Comment:
why don't you use ```docker service ps amp-integration``` ? That would not be limited to the local node. You won't get the logs but at least you can loop on it to wait for the exited status.
Another trick would be to force the node on which you execute the task, with a constraint.